control/webapp/utils.py

```diff
@@ -91,6 +92,33 @@ def parse_domain_name(domain):
         return domain.encode("idna").decode("ascii")
 
 
+def validate_domain_docroot(owner, path):
+    if not path:
+        return path, None
+    if any(ch in path for ch in string.whitespace + "\\" + '"' + "'"):
+        return path, "Document roots cannot contain spaces or quotes."
+    if path.startswith("public_html/"):
+        path = path.replace("public_html/", "", 1)
+    if isinstance(owner, Member):
+        username = owner.crsid
+        top = "home"
+    else:
+        username = owner.society
+        top = "societies"
+    base = os.path.join("/public", top, username, "public_html")
+    target = os.path.abspath(os.path.join(base, path))
+    if not target.startswith(base):
```

Comment:
I'm worried about the edge case with e.g. `path="../public_html_2"` - should this be a test with os.path.commonpath rather than a string operation?